src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java

```diff
@@ -362,6 +443,11 @@ private URL toAgentListenerURL(@Nonnull String jenkinsUrl) throws MalformedURLEx
     }
 
     public void waitForReady() throws InterruptedException {
+        if (disableHttpEndpointCheck) {
+            LOGGER.log(Level.INFO, "HTTP Endpoint Check is disabled, assuming that the target master is ready");
```

Comment:
Maybe add a sleep here to be consistent with the logic below. If the master is restarting, it's not going to be available immediately and we'd better avoid all agents from hammering the TCP port.

Reply:
How long should we sleep to be sure the master is up and running - without losing too much time unnecessarily?

Reply:
What about checking the JNLP port instead?

Reply:
I'd wait 10 seconds, as the loop below. I think in any case if the connection fails it goes back here, right?

Reply:
You're right as far as I can see.
see [#337](https://github.com/jenkinsci/remoting/pull/337/files#diff-b72206907388927fa66809bdb849f86eR452)